src/Api.php

```diff
@@ -448,4 +470,48 @@ private function getRequestParameters(array $parameters): array
         return array_merge($defaultParameters, $parameters);
     }
 
+    /**
+     * Encode a string to URL-safe base64
+     *
+     * @param string $value
+     * @return string
+     */
+    private function encodeModifiedBase64(string $value): string
+    {
+        return str_replace(['+', '/'], ['-', '_'], base64_encode($value));
+    }
+
+    /**
+     * Decode a string from URL-safe base64
+     *
+     * @param string $value
+     * @return string
+     */
+    private function decodeModifiedBase64(string $value): string
+    {
+        return base64_decode(str_replace(['-', '_'], ['+', '/'], $value));
+    }
+
+    /**
+     * Sign a URL with the current signing secret
+     *
+     * @param string $url A valid URL that is properly URL-encoded
+     * @param array $parameters Parameters to include in the URL
+     * @return string
+     */
+    public function generateSignature(string $url, ?array $parameters = null): string
+    {
+        $url = parse_url($url);
+        if (isset($parameters)) {
```

Comment:
$parameters is always set because is a parameter of this method. I think you need to replace it by:
```suggestion
        if (is_null($parameters)) {
```